### DISCUSSION TITLE
Feature ideas

### DISCUSSION THREAD

**Discussion by vtempest** [OpenSourceAGI/dev-tools-starter-agent#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by davis9001** [comment]
Hey there! 👋 

My name is David, I'm a full stack developer and I recently started primarily using SvelteKit and Cloudflare for new projects. I found this template recently and decided to kick the tires. I may either start using this as my go-to template for new projects or at least a fork of it that I modify for myself (and others) to use.

I have a few questions which I will outline in the following list:

- What plans do you have going forward where the use of Lucia is concerned (given that it was recently announced it will soon be deprecated for use as a package)?
- Are you willing to have this template converted to using a CF Pages focused approach (rather than a Workers first approach)? I can explain about this more but basically I noticed that deployment from the CLI works fine as is but some changes are necessary to get it working for deployment from a Git integration on the Cloudflare dashboard.
- Are you willing to change the name of the template to something a bit more "branding" friendly?
- What other future plans do you already have in mind for this template? I did notice it appears that Github auth is in the works and there was a slight hint of some kind of /admin routes.

Anyway I want to mention that the template is really nice so far and I appreciate the work you've put into it. Thanks for that and for accepting my first small pull request. I have been making [some more significant changes](https://github.com/davis9001/Serverless-DOCS-Stack/tree/update-enhance-fix-document-d1-and-drizle-studio) toward my goal of making the Git based deployment work without any edits along with some other things (renaming and moving configuration variables, expanding on the installation instructions, updating dependencies, fixing the Drizzle config for Studio, and more). Once I have all of that working I'll submit another pull request.

Cheers!

**Reply by vtempest** [reply]
Welcome David! 
Thanks for this and I am very interested in collaborating to make this awesome alt to Next.js

Lucia - It can be any Oauth library but this one is the best raw implementation of Google Oauth. You can try others if simple.
Workers - this has more flexibility than CF pages and there is no issue with using CLI as most do. There can perhaps be a github action for deploying it but it is not vital. CF can be a secondary option. You can make git based deploy work by using some third party api for downloading the code then uploading it.
Naming - I also have the SPOCK-DEV stack on my github projects. There can be a brand name but I feel using the acronym for the parts is best.
Todo List - I have used it in 2 projects and learned things and cleaned up, merged files. I. Docs/Wiki would be nice as well to guide people thru the filetree of the code parts.  Remove unneeded pages. Add tests. Add blog posts explaining why svelte is better than react, next js, and supabase. Make a saas boilerplate wiki like this: https://docs.opensaas.sh We can also have branches to add User Analytics, Cookie Consent, SEO Sitemap, Stripe account upgrade.
Use https://www.simpleanalytics.com which doesn't need EU cookie consent.


I appreciate your changes and we will grow this as best Svelte Saas Starter Solution. I am working on 5 contract and side projects currently and applying to accelerators.

**Reply by davis9001** [reply]
Sounds great!

Regarding Workers v Pages: I'm not sure where the extra flexibility would be with Workers over Pages but to be honest I'm still figuring all of that out myself (and I know there have been some changes with that from Cloudflare and SvelteKit). I personally have always started with a Pages first approach and still get the benefits of Workers when using the `@sveltejs/adapter-cloudflare` adapter. Perhaps we can discuss and investigate this further together at some point?

In any case the changes that I'm working on include converting to that adapter from `@sveltejs/adapter-cloudflare-workers` (not a huge change) without disturbing deployment from the CLI. This way we get the best of both worlds: Deploy via Cloudflare dashboard with Git integration or via the CLI using `wrangler` (without requiring substantial changes to  `wrangler.toml`).

I'll send you an email to the address on your GitHub profile (with my Discord info) in case you would like to chat more later.

Thanks again! :)
